### PR TITLE
chore: simplifies reconcile functions

### PR DESCRIPTION
**Pull request by bartoszmajsak**
This PR aims to simplify `Reconcile` functions used across components by removing unneeded parameters.

### CODE REVIEW

**Comment by bartoszmajsak** [comment]
@VaishnaviHire @etirelli I also have another improvement ready, but it depends on this PR. I rewrote the `reconcileSubcomponent` part to use Golang's `reflect` so new components won't need to invocations for their own instances in the main `reconcile`, you can see the code here:

https://github.com/bartoszmajsak/opendatahub-operator/commit/0b069d6e96a5adad4b61ce0ef0793d4846298375

**Comment by zdtsw** [comment]
we are making API change to deprecated "Enabled" on component.

**Comment by bartoszmajsak** [comment]
@zdtsw sure, happy to do that. Which PR should I be following to see when it lands in `main`?

**Comment by zdtsw** [comment]
DSC PR is already in, but better for another couple of hours till https://github.com/opendatahub-io/opendatahub-operator/pull/481 is in too.

**Comment by bartoszmajsak** [comment]
@zdtsw yeah noticed from the conflict list :) I will take care of rebasing once it's in too.

**Comment by bartoszmajsak** [comment]
@zdtsw  @VaishnaviHire I rebased my changes. Looking forward to your feedback.

I would also like to propose to set [`squash`](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests) option for merging PRs as the default - this way we will have a simple, linear history without merge commits. In my opinion, there is no need for this level of granularity - bringing commits that were part of the PR together with merge commit - in the history of the `main` branch. In addition we won't need to ask contributors to squash their commits, as GitHub process will take care of it automatically.

**Comment by zdtsw** [comment]
> @zdtsw @VaishnaviHire I rebased my changes. Looking forward to your feedback.
> 
> I would also like to propose to set [`squash`](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests) option for merging PRs as the default - this way we will have a simple, linear history without merge commits. In my opinion, there is no need for this level of granularity - bringing commits that were part of the PR together with merge commit - in the history of the `main` branch. In addition we won't need to ask contributors to squash their commits, as GitHub process will take care of it automatically.

I agree with the "Squash" part, if it is manual merge I usually use "Squash and Merge" button, but if it is done by ci , by default it merge with commit history. 
we can see if any objection to set the new default to "squash", anyhow, it is a good proposal!

**Comment by bartoszmajsak** [comment]
@zdtsw I noticed we started passing more data from DSCI to reconcile components (now we have both `AppNamespace` and `ManifestsUri` #463). I wonder if that would make sense to create a new config struct which will be passed around and essentially be sth like:

```golang
type DataScienceClusterConfig struct {
	*dsci.DSCInitializationSpec
}
```

Then we would avoid extending arguments further, as this could be an aggregate type for anything related to component configuration. Just a thought as this PR is intended to simplify func signatures and latest merge made it a bit more complex :) WDYT?

**Review by zdtsw — APPROVED** [review]
/lgtm

File: components/component.go [hunk 1]
```diff
@@ -21,9 +20,13 @@ type Component struct {
...
 
+func (c *Component) IsEnabled() bool {
+	return c.ManagementState == operatorv1.Managed
+}
+
 type ComponentInterface interface {
-	ReconcileComponent(owner metav1.Object, client client.Client, scheme *runtime.Scheme,
-		managementState operatorv1.ManagementState, namespace string, manifestsUri string) error
+	ReconcileComponent(cli client.Client, owner metav1.Object, namespace, manifestsUri string) error
+	IsEnabled() bool
```
Comment (etirelli) [inline components/component.go:29]: Following OpenShift best practices, we are moving to a 3-state management option for components (Managed, Unmanaged or Removed), so I am not sure introducing an IsEnabled() method into the interface is a good idea. I would suggest adding a ManagementState() method instead.

Comment (bartoszmajsak) [reply components/component.go:29]: `ManagedState` is an exported field of the base `Component` struct, so I am not sure if we really need to expose it through "getter". `IsEnabled` essentially encapsulates `managementState == operatorv1.Managed`. But perhaps we can entirely remove this and just inline this check in every component.

Comment (bartoszmajsak) [reply components/component.go:29]: Alternatively, we can just leave it as a `func` for the base `Component` which we anyway use to compose new structs but not enforce it as a contract for the interface. WDYT?

Comment (bartoszmajsak) [reply components/component.go:29]: `IsEnabled` essentially encapsulates `managementState == operatorv1.Managed` for convenience. Let me have a look.

Comment (bartoszmajsak) [reply components/component.go]: This is now addressed in f20cc74. I also rebased on top of latest `main`

**Comment by bartoszmajsak** [comment]
Would be great to close this PR as keeping up with changes popping up in the `main` branch takes time :)

File: components/codeflare/codeflare.go [hunk 2]
```diff
@@ -57,15 +56,12 @@ func (c *CodeFlare) ReconcileComponent(owner metav1.Object, cli client.Client, s
...
-		found, err := deploy.OperatorExists(cli, dependentOperator)
 
-		if !found {
-			if err != nil {
-				return err
-			} else {
-				return fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
-					dependentOperator, ComponentName)
-			}
+		if found, err := deploy.OperatorExists(cli, dependentOperator); !found {
```
Comment (bartoszmajsak) [inline components/codeflare/codeflare.go:60]: We can reduce nested `if`s by using this approach.

**Review by zdtsw — APPROVED** [review]
/lgtm

**Comment by zdtsw** [comment]
tested with local build quay.io/wenzhou/opendatahub-operator-catalog:v2.9.489
works well

**Pull request merged**